src/main/java/com/dev/springtest/dao/impl/UserDaoImpl.java

```diff
@@ -34,16 +33,10 @@ public void add(User user) {
 
     @Override
     public User getById(Long id) {
-        Transaction transaction = null;
         try (Session session = sessionFactory.openSession()) {
-            Query<User> query = session.createQuery("FROM User WHERE id = :id");
-            query.setParameter("id", id);
-            return query.getSingleResult();
+            return (User) session.createQuery("FROM User WHERE id = :id").getSingleResult();
```

Comment:
```suggestion
            return session.get(User.class, id)
```
сорі, не записалися коменти при попередній перевірці